docs/docs/jsx-in-depth.md

```diff
@@ -0,0 +1,295 @@
+
+```js
+function render1() {
+  var props = {left: 'ben', right: 'hector'};
+  return <MyComponent {...props} />;
+}
+
+function render2() {
+  return <MyComponent left="ben" right="hector" />;
+}
+```
+
+Spread attributes can be useful when you are building generic containers. However, they can also make your code messy by making it easy to pass a lot of irrelevant props to components that don't care about them. We recommend that you use this syntax sparingly.
+
+## Children in JSX
+
+In JSX expressions that contain both an opening tag and a closing tag, the content between those tags is passed as a special prop: `props.children`. There are several different ways to pass children:
+
+### String Literals
+
+You can put a string between the opening and closing tags and `props.children` will just be that string. This is useful for many of the built-in HTML elements. For example:
+
+```js
+<MyComponent>Hello world!</MyComponent>
+```
+
+This is valid JSX, and `props.children` in `MyComponent` will simply be the string `"Hello world!"`. HTML is unescaped, so you can generally write JSX just like you would write HTML in this way:
+
+```html
+<div>This is valid HTML &amp; JSX at the same time.</div>
+```
+
+### JSX Children
+
+You can provide more JSX as the children. This is useful for displaying nested components:
+
+```js
+<MyContainer>
+  <MyFirstComponent />
+  <MySecondComponent />
+</MyContainer>
+```
+
+You can mix together different types of children, so you can use string literals together with JSX children. This is another way in which JSX is like HTML, so that this is both valid JSX and valid HTML:
+
+```html
+<div>
+  Here is a list:
+  <ul>
+    <li>Item 1</li>
+    <li>Item 2</li>
+  </ul>
+</div>
+```
+
+A React component can't return multiple React elements, but a single JSX expression can have multiple children, so if you want a component to render multiple things you can wrap it in a `div` like this.
+
+When you pass JSX as the children, `props.children` does not actually let you access the child component from the parent. Instead, it lets you access the return value of `React.createElement` calls using the `React.Children` helper library. Generally, accessing children in this way is only useful to return in a render function, rather than modifying or altering the children.
+
+### JavaScript Expressions
+
+You can pass any JavaScript expression as children, by enclosing it within `{}`. For example, these expressions are equivalent:
+
+```js
+<MyComponent>foo</MyComponent>
+
+<MyComponent>{"foo"}</MyComponent>
+```
+
+This is often useful for rendering a list of JSX expressions of arbitrary length. For example, this renders an HTML list:
+
+```js
+function renderItem(message) {
+  return <li>{message}</li>;
+}
+
+function renderTodoList() {
+  var todos = ['finish doc', 'submit pr', 'nag dan to review'];
+  return (
+    <ul>
+      {todos.map(renderItem)}
```

Comment:
An inquisitive reader would think: when don't want to do this instead of using `<Item>`?

In general we want to encourage people to create components rather than call functions directly. But I guess this is fine.

I don't have a good solution to this.